distributed/protocol/compression.py

```diff
@@ -124,14 +124,16 @@ def byte_sample(b, size, n):
     n : int
         number of samples to collect
     """
+    b = ensure_memoryview(b)
+
     starts = [random.randint(0, len(b) - size) for j in range(n)]
     ends = []
     for i, start in enumerate(starts[:-1]):
         ends.append(min(start + size, starts[i + 1]))
     ends.append(starts[-1] + size)
 
     parts = [b[start:end] for start, end in zip(starts, ends)]
-    return b"".join(map(ensure_bytes, parts))
+    return b"".join(parts)
```

Comment:
Special case where there is only one part?

Reply:
Yep this is easy to do

```suggestion
    if n == 0:
        return b""
    elif n == 1:
        return parts[0]
    else:
        return b"".join(parts)
```